tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.Dump.cs

```diff
@@ -0,0 +1,289 @@
+
+        var breakLabel = Expression.Label("loopBreak");
+        var loopBodyExpression = Expression.IfThenElse(
+            Expression.Equal(moveNextCall, Expression.Constant(true)),
+            Expression.Block(
+                new[] { loopItem },
+                Expression.IfThenElse(
+                    Expression.LessThan(index, Expression.Constant(3)),
+                    Expression.Block(
+                        Expression.IfThen(
+                            Expression.GreaterThan(index, Expression.Constant(0)),
+                            Expression.Call(result, stringBuilderAppend, Expression.Constant(", "))),
+                        Expression.Assign(loopItem, Expression.Property(enumeratorVar, "Current")),
+                        dumpObjectCallExpression,
+                        Expression.PostIncrementAssign(index)),
+                    Expression.Block(
+                        Expression.Call(result, stringBuilderAppend, Expression.Constant(", ...")),
+                        Expression.Break(breakLabel)))),
+            Expression.Break(breakLabel));
+
+        expressions.Add(Expression.Loop(loopBodyExpression, breakLabel));
+        expressions.Add(Expression.Call(result, stringBuilderAppend, Expression.Constant("]")));
+        expressions.Add(Expression.Call(result, GetMethodByReflection(typeof(StringBuilder), nameof(StringBuilder.ToString), Type.EmptyTypes)));
+        return Expression.Block(expressions);
+    }
+
+    private Expression DumpFieldsExpression(Expression expression, List<ParameterExpression> scopeMembers)
+    {
+        const BindingFlags flags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance | BindingFlags.DeclaredOnly;
+        var getTypeMethod = GetMethodByReflection(typeof(object), nameof(object.GetType), Type.EmptyTypes);
+        var getFieldsMethod = GetMethodByReflection(typeof(Type), nameof(Type.GetFields), new[] { typeof(BindingFlags) });
+        var getFields = Expression.Call(Expression.Call(expression, getTypeMethod), getFieldsMethod, Expression.Constant(flags));
+        var stringBuilderAppend = GetMethodByReflection(typeof(StringBuilder), nameof(StringBuilder.Append), new[] { typeof(string) });
+        var expressions = new List<Expression>();
+
+        var fields = Expression.Variable(typeof(FieldInfo[]), "fieldsArray");
+        scopeMembers.Add(fields);
+        expressions.Add(Expression.Assign(fields, getFields));
+
+        var result = Expression.Variable(typeof(StringBuilder), "fieldValues");
+        scopeMembers.Add(result);
+        expressions.Add(Expression.Assign(result, Expression.New(typeof(StringBuilder).GetConstructor(Type.EmptyTypes))));
+
+        var index = Expression.Variable(typeof(int), "index");
+        scopeMembers.Add(index);
+        expressions.Add(Expression.Assign(index, Expression.Constant(0)));
+
+        // Loop Content
+        var fieldAtIndex = Expression.ArrayIndex(fields, index);
```

Comment:
Seems like this would be much easier to understand as "real" code instead of an expression that's being constructed. Is there any reason not to create a `DumpFields` method that does the equivalent of this logic?

Reply:
The logic can be in a "real" method but we need to pass an object to work on and we have only expression in our hand. I can try to get the real object from the scope members but I'm not sure what is better. WDYT?

Reply:
I'm ok with leaving it as-is then.